### PR TITLE
 Add Object Cache panel to Query Monitor

### PR DESCRIPTION
**Pull request by trepmal**
## Description

Adds panel to Query Monitor that shows Object Cache stats. This the same data as currently shown in Debug Bar.

## Changelog Description

Show Object Cache stats to Query Monitor

## Checklist

Please make sure the items below have been covered before requesting a review:

- [x] This change works and has been tested locally (or has an appropriate fallback).
- [x] This change works and has been tested on a Go sandbox.
- [ ] This change has relevant unit tests (if applicable).
- [ ] This change has relevant documentation additions / updates (if applicable).
- [ ] I've created a changelog description that aligns with the provided examples.

## Steps to Test

Install and view:

qm-object-cache

Enable Debug Bar and notice "Memcache" item not shown while QM is active:

debug-bar-no-cache

### CODE REVIEW

**Comment by rinatkhaziev** [comment]
This makes me extremely happy

File: qm-plugins/qm-object-cache/qm-object-cache.php [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+		$collector = QM_Collectors::get( 'object_cache' );
+		if ( $collector ) {
+			$output['object_cache'] = new QM_Output_ObjectCache( $collector );
+		}
+		return $output;
+	}, 101, 2 );
+} );
+
+function qm_object_cache_assets() {
+	wp_enqueue_style( 'qm-objectcache-style', plugin_dir_url( __FILE__ ) . 'css/style.css', null, '0.1' );
```
Comment (sjinks) [inline qm-plugins/qm-object-cache/qm-object-cache.php:36]: A minor nitpick:

```suggestion
	wp_enqueue_style( 'qm-objectcache-style', plugin_dir_url( __FILE__ ) . 'css/style.css', array(), '0.1' );
```

According to the [documentation](https://developer.wordpress.org/reference/functions/wp_enqueue_style/), the third parameter is an array of strings, so let's keep it that way, please.

File: qm-plugins/qm-object-cache/class-qm-output-object-cache.php [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+		add_filter( 'qm/output/menu_class', array( $this, 'admin_class' ) );
+		add_filter( 'qm/output/menus', array( $this, 'admin_menu' ), 101 );
+	}
+
+	/**
+	 * Outputs data in the footer
+	 */
+	public function output() {
+		?>
+		<div class="qm qm-non-tabular" id="<?php echo esc_attr( $this->collector->id() ); ?>">
```
Comment (sjinks) [inline qm-plugins/qm-object-cache/class-qm-output-object-cache.php:21]: I see that this `<div>` is not closed in this method. I just want to confirm that this is OK.

File: qm-plugins/qm-object-cache/qm-object-cache.php [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+ * as the output contains ID attributes.
+ */
+add_filter( 'debug_bar_panels', function( $panels ) {
+	foreach ( $panels as $k => $panel ) {
+		if ( is_a( $panel, 'Debug_Bar_Object_Cache' ) ) {
+			unset( $panels[ $k ] );
+		}
+	}
+	$panels = array_values( $panels ); // need to reset keys or Debug Bar throws warning
+	return $panels;
```
Comment (sjinks) [inline qm-plugins/qm-object-cache/qm-object-cache.php:50]:
```suggestion
    $panels = array_filter( $panels, fn( $panel ) => ! is_a( $panel, Debug_Bar_Object_Cache::class ) );
    return array_values( $panels );
```

I won't insist on this change, though :-)

**Review by sjinks — CHANGES_REQUESTED** [review]
Looks awesome! I have left a few comments.

File: qm-plugins/qm-object-cache/class-qm-output-object-cache.php [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+		<div id="object-cache-stats">
+			<?php
+			global $wp_object_cache;
+			$wp_object_cache->stats();
+			?>
+		</div>
```
Comment (trepmal) [inline qm-plugins/qm-object-cache/class-qm-output-object-cache.php:27]:
```suggestion
		</div>
		</div>
```

**Review by sjinks — APPROVED** [review]

**Comment by sjinks** [comment]
r2491-stacks

**Pull request merged**